src/cpu/x64/asm.rs

```diff
@@ -2091,6 +2123,26 @@ mod tests {
         assert_emit!(0xf2, 0x49, 0x0f, 0x2c, 0xe0; cvttsd2si(1, RSP, XMM8));
     }
 
+    #[test]
+    fn test_movd_f2i_32() {
```

Comment:
I think we need only two test functions: `test_movd_f2i` and `test_movd_i2f`. Can you add a bit more tests with different registers, e.g. R8, R15, XMM7, XMM8, XMM15. Only like two or three combinations. It would be good to test them as well since those registers also set a bit in the rex-byte (register index is greater than 7 and therefore bit 3 is non 0)